frontend/src/api/actions/init.js

```diff
@@ -0,0 +1,150 @@
+import {
+    setActiveSession,
+    // setActiveUserRole,
+    fetchActiveUser,
+    fetchSessions,
+    fetchApplicants,
+    fetchApplications,
+    fetchAssignments,
+    fetchContractTemplates,
+    fetchInstructors,
+    fetchPositions
+} from "./";
+import { setGlobals, globalsSelector } from "../../views/globals/actions";
+
+const toggleMockApi = (getState, newMockAPIValue) => {
+    // in production, ToggleMockApi is a no-op. In development, it actually
+    // does something.
+    /* eslint-disable */
+    if (newMockAPIValue != null && process.env.REACT_APP_DEV_FEATURES) {
+        const mock = require("../mockAPI");
+        const mockAPI = mock.mockAPI;
+        const state = getState();
+
+        if (newMockAPIValue === true) {
+            mockAPI.replaceGlobalFetch();
+        } else {
+            mockAPI.restoreGlobalFetch();
+        }
+    }
+    /* eslint-enable */
+};
+
+const updateGlobals = (getState, newMockAPIValue, newActiveSession) => {
+    const state = getState();
+    let globals = globalsSelector(state);
+
+    if (newMockAPIValue != null) {
+        globals = {
+            ...globals,
+            mockAPI: newMockAPIValue
+        };
+    }
+
+    if (newActiveSession) {
+        globals = {
+            ...globals,
+            activeSession: newActiveSession.id
+        };
+    }
+
+    return setGlobals(globals);
+};
+
+const generateListOfActions = (getState, stage, options) => {
```

Comment:
Make this a `function` function. This needs some comments.

Reply:
The order of arguments is strange. Look at where `getState` appears in other actions.